### PR TITLE
Time series - POST transformation

### PR DESCRIPTION
**Pull request by marcel2012**
example requests
```json
{
  "name": "quadrants",
  "source_time_series_ids": [
    0,
    7
  ],
  "destination_observable_information_id": 0,
  "destination_measure_id": 0,
  "additional_properties": [
    {
      "key": "origin_y",
      "value": 3
    },
    {
      "key": "origin_x",
      "value": 15
    }
  ]
}
```
```json
{
  "name": "resample_nearest",
  "source_time_series_ids": [
    6
  ],
  "destination_observable_information_id": 0,
  "destination_measure_id": 0,
  "additional_properties": [
    {
      "key": "period",
      "value": 50
    },
    {
      "key": "some_other_param",
      "value": 15
    }
  ]
}
```

### CODE REVIEW

File: grisera_api/time_series/transformation/TimeSeriesTransformationFactory.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+        """
+        Transform time series data
+
+        Args:
+            transformation_name (str): Name of transformation
+
+        Returns:
+            New time series transformation class
+        """
+        if transformation_name == "resample_nearest":
```
Comment (mrwrob) [inline grisera_api/time_series/transformation/TimeSeriesTransformationFactory.py:22]: It'd be a good idea to create enum class for these name, something like:
```
class TransformationTypes(Enum):
    RESAMPLE_NEAREST = 1
    QUADRANTS = 2
```
or
```
class TransformationTypes(Enum):
    RESAMPLE_NEAREST = "resample_nearest"
    QUADRANTS = "quadrants"
```

File: grisera_api/time_series/transformation/TimeSeriesTransformationQuadrants.py [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+            New time series object
+        """
+        assert len(time_series) == 2, "Number of time series should equals 2 for quadrants transformation"
+        assert time_series[0].type == time_series[1].type, "Time series types should be equal"
+        origin_x = get_additional_parameter(additional_properties, "origin_x")
+        origin_y = get_additional_parameter(additional_properties, "origin_y")
+        if origin_x is None:
+            origin_x = 0
+        else:
+            origin_x = int(origin_x)
```
Comment (mrwrob) [inline grisera_api/time_series/transformation/TimeSeriesTransformationQuadrants.py:33]: It'd be more pythonic way to use:
```
origin_x = 0 if origin_x is None else origin_x = int(origin_x)
```

File: grisera_api/time_series/transformation/TimeSeriesTransformationQuadrants.py [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+        timestamp_label = "timestamp" if time_series[0].type == Type.timestamp else "start_timestamp"
+        for current_signal_value_x_index in range(len(time_series[0].signal_values)):
+            while current_signal_value_y_index < len(time_series[1].signal_values) and \
+                    int(get_node_property(time_series[1].signal_values[current_signal_value_y_index]["timestamp"],
+                                          timestamp_label)) < int(
+                get_node_property(time_series[0].signal_values[current_signal_value_x_index]["timestamp"],
+                                  timestamp_label)):
+                current_signal_value_y_index += 1
+            if current_signal_value_y_index < len(time_series[1].signal_values) and \
+                    get_node_property(time_series[0].signal_values[current_signal_value_x_index]["timestamp"],
```
Comment (mrwrob) [inline grisera_api/time_series/transformation/TimeSeriesTransformationQuadrants.py:55]: Please add comments to describe  these conditions. They are very hard to read.

File: grisera_api/time_series/transformation/TimeSeriesTransformationResample.py [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+        if start_timestamp is None:
+            start_timestamp = 0
+        else:
+            start_timestamp = int(start_timestamp)
+        if end_timestamp is not None:
+            end_timestamp = int(end_timestamp)
+        else:
+            end_timestamp = period + int(
+                get_node_property(time_series[0].signal_values[-1]["timestamp"], end_timestamp_label))
+
```
Comment (mrwrob) [inline grisera_api/time_series/transformation/TimeSeriesTransformationResample.py:43]: Please use conditional expression `<var> = <expr1> if <conditional_expr> else <expr2>`

File: grisera_api/time_series/transformation/TimeSeriesTransformationResample.py [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+        new_signal_values = []
+        new_signal_values_index_mapping = []
+        current_time = start_timestamp
+        current_signal_value_index = 0
+        while current_time < end_timestamp:
+            while current_signal_value_index + 1 < len(time_series[0].signal_values) and \
+                    int(get_node_property(time_series[0].signal_values[current_signal_value_index]["timestamp"],
+                                          end_timestamp_label)) <= current_time:
+                current_signal_value_index += 1
```
Comment (mrwrob) [inline grisera_api/time_series/transformation/TimeSeriesTransformationResample.py:56]: Please add comments to describe the conditions

**Comment by mrwrob** [comment]
Well done, just a couple of remarks.

**Review by mrwrob — CHANGES_REQUESTED** [review]
Well done, just a couple of remarks.

File: grisera_api/time_series/transformation/TimeSeriesTransformationResample.py (hunk 5 above)
Comment (marcel2012) [reply grisera_api/time_series/transformation/TimeSeriesTransformationResample.py]: I added comments. Is it clear now?

**Review by mrwrob — APPROVED** [review]
Great! Thank you!

**Review by terzawad — APPROVED** [review]

**Pull request merged**